luarules/gadgets/cmd_build_bugger_off.lua

```diff
@@ -0,0 +1,204 @@
+local gadget = gadget ---@type gadget
+
+function gadget:GetInfo()
+    return {
+        name    = "Builder buggeroff",
+        desc    = "Enables busy builders and moving units to buggeroff",
+        author  = "Flameink",
+        date    = "March 7, 2025",
+        version = "1.0",
+        license = "GNU GPL, v2 or later",
+        layer   = 0,
+        enabled = true   --  loaded by default?
+    }
+end
+
+local debugLog = false
+
+function print(Message)
+   if debugLog then
+        Spring.Echo(Message)
+    end
+end
+
+local shouldNotBuggeroff = {}
+local cachedUnitDefs = {}
+for unitDefID, unitDef in pairs(UnitDefs) do
+    if unitDef.speed == 0 or unitDef.isBuilding or unitDef.isFactory then
+        shouldNotBuggeroff[unitDefID] = true
+    end
+    cachedUnitDefs[unitDefID] = { radius = unitDef.radius, isBuilder = unitDef.isBuilder}
+end
+
+if gadgetHandler:IsSyncedCode() then
+
+    local function closestPointOnCircle(cx, cz, radius, tx, tz)
+        local dx = tx - cx
+        local dz = tz - cz
+        local dist = math.diag(dx, dz)
+        if dist == 0 then
+            -- Target is exactly at center; choose arbitrary point on circle
+            return cx + radius, cz
+        end
+        local scale = radius / dist
+        local closestX = cx + dx * scale
+        local closestZ = cz + dz * scale
+        return closestX, closestZ
+    end
+
+    local function WillBeNearTarget(unitID, tx, ty, tz, frames, maxDistance)
+        local ux, uy, uz = Spring.GetUnitPosition(unitID)
+        if not ux then return false end
+        
+        local vx, vy, vz = Spring.GetUnitVelocity(unitID)
+        if not vx then return false end
+        
+        -- Predict future position
+        local futureX = ux + vx * frames
```

Comment:
Consider making the function accept seconds instead of frames to make it independent of frame rate:
```suggestion
        local futureX = ux + vx * seconds * Game.gameSpeed
```
This would also involve changing `BUILDER_DELAY_FRAME_COUNT` below to be in seconds (100 frames is 3⅓ seconds).